src/app/clusters/laundry-washer-controls-server/laundry-washer-controls-server.h

```diff
@@ -0,0 +1,57 @@
+/**
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace LaundryWasherControls {
+
+class LaundryWasherManager
+{
+public:
+    template <typename T>
+    struct AttributeProvider
+    {
+        using pointer = const T *;
+
+        inline pointer begin() const { return mBegin; }
+        inline pointer end() const { return mEnd; }
+
+        AttributeProvider() : mBegin(nullptr), mEnd(nullptr) {}
+
+        AttributeProvider(const pointer aBegin, const pointer aEnd) : mBegin(aBegin), mEnd(aEnd) {}
```

Comment:
This setup requires the whole array to be in RAM at once, right?  And how does the thing creating the provider know when it's safe to deallocate the array?

I strongly suggest writing an implementation of LaundryWasherManager that does not use "always in RAM" static data to back it (could pretend like you are reading data from flash as needed, for purposes of seeing how thing hang together), and seeing whether that works with this API.

Reply:
I will add const to the attributes

Reply:
Adding const is not going to address the problems above, unless I totally mis-understand what you mean.

Reply:
Could you share an example for our reference? Thank you

Reply:
Something like the Operational State cluster could be an example of this sort of provider....

Reply:
Thank you. Let me check

Reply:
Please help to check again. Thank you